File: src/Controls/KarveControls/DataGrid/KarveGridTest/KarveGrid/GridOrder/DataGridOrdenColumna.cs
namespace KarveGrid.GridOrder
{
    public class DataGridOrdenColumna
    {

        public enum euCriterio
        {
            Asc,
            Desc
        }

        //*
        euCriterio _eCriterio;
        //*
        int _Item;
        //*
        string _sName;
        string _sCampo;
        //*
        string _sExpresion;
        //*
        string _sExpresionBD;
        //*
        string _sCriterio;
        //*
        string _sAliasTabla;
        //*
        string _sAliasCampo;

        public int Item {
            get { return _Item; }
            set { _Item = value; }
        }

        public string Name {
            get { return _sName; }
            set { _sName = value; }
        }

        public string Expresion {
            get { return _sExpresion; }
            set { _sExpresion = value; }
        }

        public string ExpresionBD {
            get { return _sExpresionBD; }
            set { _sExpresionBD = value; }
        }

        public string AliasTabla {
            get { return _sAliasTabla; }
            set { _sAliasTabla = value; }
        }

        public euCriterio Criterio {
            get { return _eCriterio; }
            set {
                _eCriterio = value;
                switch (_eCriterio) {
                    case euCriterio.Asc:
                        _sCriterio = " ASC ";
                        break;
                    case euCriterio.Desc:
                        _sCriterio = " DESC ";
                        break;
                }
            }
        }

        public string CriterioString {
            get { return _sCriterio; }
        }

        public string Campo {
            get { return _sCampo; }
            set { _sCampo = value; }
        }

        public string AliasCampo {
            get { return _sAliasCampo; }
            set { _sAliasCampo = value; }
        }

    }
}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================

File: documentation/Karve/CustomControls/DataGrid/Clases/DataGridOrder/DataGridOrdenColumna.vb
Public Class DataGridOrdenColumna

    Public Enum euCriterio
        Asc
        Desc
    End Enum

    Dim _eCriterio As euCriterio        '*
    Dim _Item As Integer                '*
    Dim _sName As String                '*
    Dim _sCampo As String
    Dim _sExpresion As String           '*
    Dim _sExpresionBD As String           '*
    Dim _sCriterio As String            '*
    Dim _sAliasTabla As String          '*
    Dim _sAliasCampo As String          '*

    Public Property Item As Integer
        Get
            Return _Item
        End Get
        Set(value As Integer)
            _Item = value
        End Set
    End Property

    Public Property Name As String
        Get
            Return _sName
        End Get
        Set(value As String)
            _sName = value
        End Set
    End Property

    Public Property Expresion As String
        Get
            Return _sExpresion
        End Get
        Set(value As String)
            _sExpresion = value
        End Set
    End Property

    Public Property ExpresionBD As String
        Get
            Return _sExpresionBD
        End Get
        Set(value As String)
            _sExpresionBD = value
        End Set
    End Property

    Public Property AliasTabla As String
        Get
            Return _sAliasTabla
        End Get
        Set(value As String)
            _sAliasTabla = value
        End Set
    End Property

    Public Property Criterio As euCriterio
        Get
            Return _eCriterio
        End Get
        Set(value As euCriterio)
            _eCriterio = value
            Select Case _eCriterio
                Case euCriterio.Asc : _sCriterio = " ASC "
                Case euCriterio.Desc : _sCriterio = " DESC "
            End Select
        End Set
    End Property

    Public ReadOnly Property CriterioString As String
        Get
            Return _sCriterio
        End Get
    End Property

    Property Campo As String
        Get
            Return _sCampo
        End Get
        Set(value As String)
            _sCampo = value
        End Set
    End Property

    Public Property AliasCampo As String
        Get
            Return _sAliasCampo
        End Get
        Set(value As String)
            _sAliasCampo = value
        End Set
    End Property

End Class
